File: Sources/ClipDeckKit/Model/Folder.swift
import Foundation

/// Uma pasta, agrupando templates por objetivo (Prospecção, Cobrança, ...).
public struct Folder: Codable, Identifiable, Equatable, Hashable {
    public let id: UUID
    public var name: String
    public var symbol: String
    public var order: Int

    public init(id: UUID = UUID(), name: String, symbol: String = "folder", order: Int = 0) {
        self.id = id
        self.name = name
        self.symbol = symbol
        self.order = order
    }

    public init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id = try c.decodeIfPresent(UUID.self, forKey: .id) ?? UUID()
        name = try c.decodeIfPresent(String.self, forKey: .name) ?? "Sem nome"
        symbol = try c.decodeIfPresent(String.self, forKey: .symbol) ?? "folder"
        order = try c.decodeIfPresent(Int.self, forKey: .order) ?? 0
    }
}

File: windows/src/ClipDeck.Core/Models/Folder.cs
namespace ClipDeck.Core.Models;

/// <summary>Uma pasta, agrupando templates por objetivo.</summary>
public sealed class Folder
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = "Sem nome";

    /// <summary>Nome do glifo usado na interface.</summary>
    public string Symbol { get; set; } = "folder";

    public int Order { get; set; }
}
